eng/common/scripts/Submit-AzureSdkForNetPr.ps1

```diff
@@ -0,0 +1,223 @@
+        throw "eng/Packages.Data.props not found in the repository"
+    }
+
+    $propsFileContent = Get-Content $propsFilePath -Raw
+    
+    # Update the UnbrandedGeneratorVersion property in the file
+    $pattern = '<UnbrandedGeneratorVersion>[^<]*</UnbrandedGeneratorVersion>'
+    $replacement = '<UnbrandedGeneratorVersion>' + $PackageVersion + '</UnbrandedGeneratorVersion>'
+    
+    $updatedContent = $propsFileContent -replace $pattern, $replacement
+    
+    if ($updatedContent -eq $propsFileContent) {
+        Write-Warning "No changes were made to eng/Packages.Data.props. The UnbrandedGeneratorVersion property might not exist or have a different format."
+        Write-Host "Current content around UnbrandedGeneratorVersion:"
+        $propsFileContent | Select-String -Pattern "UnbrandedGeneratorVersion" -Context 2, 2
+    }
+    
+    # Write the updated file back
+    Set-Content -Path $propsFilePath -Value $updatedContent -NoNewline
+
+    # Update the dependency in eng/packages/http-client-csharp/package.json
+    Write-Host "Updating dependency version in eng/packages/http-client-csharp/package.json..."
+    $packageJsonPath = Join-Path $tempDir "eng/packages/http-client-csharp/package.json"
+    
+    if (-not (Test-Path $packageJsonPath)) {
+        throw "eng/packages/http-client-csharp/package.json not found in the repository"
+    }
+
+    $packageJsonContent = Get-Content $packageJsonPath -Raw | ConvertFrom-Json
+    
+    # Update the Microsoft.TypeSpec.Generator.ClientModel dependency version
+    $updated = $false
+    if ($packageJsonContent.dependencies -and $packageJsonContent.dependencies."@typespec/http-client-csharp") {
+        $packageJsonContent.dependencies."@typespec/http-client-csharp" = $PackageVersion
+        $updated = $true
+        Write-Host "Updated @typespec/http-client-csharp in dependencies"
+    }
+    if ($packageJsonContent.devDependencies -and $packageJsonContent.devDependencies."@typespec/http-client-csharp") {
+        $packageJsonContent.devDependencies."@typespec/http-client-csharp" = $PackageVersion
+        $updated = $true
+        Write-Host "Updated @typespec/http-client-csharp in devDependencies"
+    }
+    if ($packageJsonContent.peerDependencies -and $packageJsonContent.peerDependencies."@typespec/http-client-csharp") {
+        $packageJsonContent.peerDependencies."@typespec/http-client-csharp" = $PackageVersion
+        $updated = $true
+        Write-Host "Updated @typespec/http-client-csharp in peerDependencies"
+    }
+    
+    if (-not $updated) {
+        Write-Warning "No @typespec/http-client-csharp dependency found in package.json"
+    } else {
+        # Write the updated package.json back
+        $packageJsonContent | ConvertTo-Json -Depth 10 | Set-Content -Path $packageJsonPath
+        
+        # Run npm install in the http-client-csharp directory
+        Write-Host "Running npm install in eng/packages/http-client-csharp..."
```

Comment:
After running npm install. We should run `npm run build` and then eng/scripts/Generate.ps1.